src/hyperdeck/hyperdeck-core.js

```diff
@@ -12,10 +12,34 @@ var logger = Logger.get('hyperdeck.HyperdeckCore');
  * Allows you to make requests to the hyperdeck and get its parsed responses.
  * This chains the requests so only one is sent at a time.
  * You can also listen for asynchronous events sent from the hyperdeck.
- * @param ip, The IP address of the hyperdeck.
+ * @param config hyperdeck configuration
+ *               config = 'ip address string[string]'
+ *               or
+ *               config = {
+ *                 ip: 'ip address string[string]',
+ *                 [ pingInterval: ping interval in miliseconds[int][default = 10000] ],
+ *                 [ timeout: inactivity timeout in miliseconds[int][default = 3000] ]
+ *               }
  **/
-function HyperdeckCore(ip) {
+function HyperdeckCore(config) {
+  if (config === null) {
+    throw 'Invalid Configuration, please refer to documentations.';
+  }
+
   var self = this;
+  var cfg = {};
```

Comment:
This can be a bit neater using Object.assign

var cfg = Object.assign({ pingInterval: 10000 }, config)

Where you would check if the provided config is a string and set it to { ip: config } above if it is 

This also means we never mutate the input :)

Reply:
```
var cfg = {};
if (typeof config === 'string') {
  cfg.ip = config;
}
cfg = Object.assign({ pingInterval: 10000, timeout: 3000 }, cfg);
```

Do you mean like this?

Reply:
Almost 

if (typeof config === 'string') {
  config = { ip: config.ip };
}
config = Object.assign({ pingInterval: 10000, timeout: 3000 }, config);

Reply:
you mean:

```
if (typeof config === 'string') {
config = { ip: config };
}
config = Object.assign({ pingInterval: 10000, timeout: 3000 }, config);
```

yes its better